app/main/views/frameworks.py

```diff
@@ -273,6 +273,73 @@ def framework_start_supplier_declaration(framework_slug):
                            framework_close_date=framework_close_date), 200
 
 
+@main.route('/frameworks/<framework_slug>/declaration/reuse', methods=['GET'])
+@login_required
+def reuse_framework_supplier_declaration(framework_slug):
+    """Attempt to find a supplier framework declaration that we can reuse.
+
+    :param: framework_slug
+    :query_param: reusable_declaration_framework_slug
+    :return: 404, redirect or reuse page (frameworks/reuse_declaration.html)
+    """
+    reusable_declaration_framework_slug = request.args.get('reusable_declaration_framework_slug', None)
+    supplier_id = current_user.supplier_id
+    # Get the current framework.
+    current_framework = data_api_client.get_framework(framework_slug)['frameworks']
+    if reusable_declaration_framework_slug:
+        # If they have supplied a declaration then force it. This is an 'admin' feature.
+        # Experimental feature to be used to force through reuse.
+        try:
+            # Get their old declaration.
+            declaration = data_api_client.get_supplier_declaration(supplier_id, reusable_declaration_framework_slug)
+            old_framework = data_api_client.get_framework(reusable_declaration_framework_slug)['frameworks']
+        except APIError:
+            abort(404)
+    else:
+        # Otherwise then attempt to determine if they have a declaration we can offer for reuse.
+        declarations = data_api_client.find_supplier_declarations(supplier_id)['frameworkInterest']
```

Comment:
I'd be tempted to push this API call to get `declarations` into the `get_reusable_declaration` method itself - the client is passed in there anyway.  It feels a bit weird to get the list of declarations here and then have the logic to turn the list into a dict in the method - that would be nicer done all together.